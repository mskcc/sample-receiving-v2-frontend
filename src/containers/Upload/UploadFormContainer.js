import React, { Component } from 'react'

import PropTypes from 'prop-types'

import { connect } from 'react-redux'
import { uploadFormActions } from '../../actions'

import { UploadForm } from '../../components/Upload'

export class UploadFormContainer extends React.Component {
  constructor(props) {
    super(props)
  }

  componentDidUpdate(prevProps, prevState) {}

  componentDidMount() {
    this.props.getInitialState()
  }
  // handleSubmit = formValues => {
  // this.props.handleSubmit(formValues)
  // }

  handleMaterialChange = selectedMaterial => {
    if (selectedMaterial) {
      // get possible applications for this material
      this.props.getApplicationsForMaterial(selectedMaterial)
    } else {
      this.props.clearMaterial()
    }
    // show containers depending on material combination
    this.props.filterContainers(selectedMaterial)
  }

  handleApplicationChange = selectedApplication => {
    if (selectedApplication) {
      // get possible ,materials for this application
      this.props.getMaterialsForApplication(selectedApplication)
    } else {
      this.props.clearApplication()
    }
  }

  handleSpeciesChange = selectedSpecies => {
    if (selectedSpecies) {
      this.props.getFormatterForSpecies(selectedSpecies)
    } else this.props.clearSpecies()
  }

  render() {
    const {
      classes,
      form,
      handleSubmit,
      gridIsLoading,
      nothingToChange,
    } = this.props
    return (
      <UploadForm
        form={form}
        gridIsLoading={gridIsLoading}
        nothingToChange={nothingToChange}
        handleSubmit={handleSubmit}
        handleMaterialChange={this.handleMaterialChange}
        handleApplicationChange={this.handleApplicationChange}
        handleSpeciesChange={this.handleSpeciesChange}
      />
    )
  }
}

UploadFormContainer.defaultProps = {
  getInitialState: () => {},
}

const mapStateToProps = state => ({
  form: state.upload.form,
})

const mapDispatchToProps = {
  ...uploadFormActions,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UploadFormContainer)
