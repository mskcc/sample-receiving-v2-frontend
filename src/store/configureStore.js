import { createStore, applyMiddleware, compose } from 'redux'
import { Provider } from 'react-redux'

import thunk from 'redux-thunk'
import multi from 'redux-multi'
import { persistStore } from 'redux-persist'
import { createLogger } from 'redux-logger'
import rootReducer from '../reducers/rootReducer'
import DevTools from '../containers/DevTools'

const middleware = [thunk, multi]
if (process.env.NODE_ENV === 'production') {
}

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(...middleware),
    DevTools.instrument()
  )
)


let persistor = persistStore(store)

export {store, persistor}
