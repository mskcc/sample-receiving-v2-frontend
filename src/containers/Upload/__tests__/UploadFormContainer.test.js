import React, { Component } from 'react'

import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import multi from 'redux-multi'

import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { UploadFormContainer } from '../UploadFormContainer'

import { Provider } from 'react-redux'

// import { filledFullStateMock } from '../../../mocks'

// const formTestStore = initialFormStateMock

const middlewares = [thunk, multi]
const mockStore = configureStore(middlewares)

// const UploadFormContainer = props => <TestUploadFormContainer />

describe('Render UploadFormContainer', () => {
  it('render UploadFormContainer correctly', () => {
    const UploadFormContainerComponent = renderer
      .create(<UploadFormContainer />)
      .toJSON()
    expect(UploadFormContainerComponent).toMatchSnapshot()
  })
})

describe('does sth', () => {
  it('has UploadForm', () => {
    const getInitialState = jest.fn()
    const wrapper = mount(
      <UploadFormContainer getInitialState={getInitialState} />
    )
    const instance = wrapper.instance()
    expect(wrapper.find('UploadForm').exists()).toBe(false)
    //TODO broke when form ? : UploadForm was introduced for production expect(wrapper.find('UploadForm').exists()).toBe(true)
    expect(getInitialState).toHaveBeenCalled()
  })

  it('handles species change', () => {
    const clearSpecies = jest.fn()
    const getFormatterForSpecies = jest.fn()
    const wrapper = mount(
      <UploadFormContainer
        getFormatterForSpecies={getFormatterForSpecies}
        clearSpecies={clearSpecies}
      />
    )
    const instance = wrapper.instance()
    instance.handleSpeciesChange('human')
    expect(getFormatterForSpecies).toHaveBeenCalled()
    instance.handleSpeciesChange()
    expect(clearSpecies).toHaveBeenCalled()
  })

  it('handles material change', () => {
    const clearMaterial = jest.fn()
    const getApplicationsForMaterial = jest.fn()
    const filterContainers = jest.fn()

    const wrapper = mount(
      <UploadFormContainer
        clearMaterial={clearMaterial}
        getApplicationsForMaterial={getApplicationsForMaterial}
        filterContainers={filterContainers}
      />
    )
    const instance = wrapper.instance()
    instance.handleMaterialChange('cells')
    expect(getApplicationsForMaterial).toHaveBeenCalled()
    expect(filterContainers).toHaveBeenCalled()
    instance.handleMaterialChange()
    expect(clearMaterial).toHaveBeenCalled()
    expect(filterContainers).toHaveBeenCalled()
  })

  it('handles application change', () => {
    const clearApplication = jest.fn()
    const getMaterialsForApplication = jest.fn()

    const wrapper = mount(
      <UploadFormContainer
        getMaterialsForApplication={getMaterialsForApplication}
        clearApplication={clearApplication}
      />
    )
    const instance = wrapper.instance()
    instance.handleApplicationChange('AmpliSeq')
    expect(getMaterialsForApplication).toHaveBeenCalled()
    instance.handleApplicationChange()
    expect(clearApplication).toHaveBeenCalled()
  })
})
