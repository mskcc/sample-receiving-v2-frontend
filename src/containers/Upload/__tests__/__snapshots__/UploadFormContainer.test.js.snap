// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UploadFormContainer render UploadFormContainer correctly 1`] = `<div />`;
